### DISCUSSION TITLE
Assign bug reports an effort/difficulty rating and allowing subscribers to vote on bug reports to fix

### DISCUSSION THREAD

**Discussion by andovcr** [chromiecraft/chromiecraft#7122] [Suggestions]
I know the CC staff do not have a favorable opinion of "prioritizing" bug reports, but I wonder if they would be open to assigning a effort/difficulty rating (say 1-4 with 1 being trivial and 4 being extremely intensive).  At a glance, this would give the wider player base some visibility into how much work is required.  I understand that it would be time-consuming to assign every bug a rating, but at the same time, staff have previously said that they are constantly aware of what's submitted and even if only half of the bugs receive a rating, then that's better than nothing (don't let perfect be the enemy of good).

If staff were to commit to this, and I'm sure it's a big IF, then they could integrate it with the subscriptions program to allow for enhanced community engagement and a direct feedback loop that I believe is currently missing.

Example:  Each month,
Tier 1 subscribers would get to vote for one rank 1 bug to be fixed.
Tier 2 subscribers would get to vote for one rank 2 bug and one rank 1 bug to be fixed.
Tier 3 subscribers would get to vote for one rank 3 bug, one rank 2 bug, and one rank 1 bug to be fixed.

Resulting in three rank 1 bugs, two rank 2 bugs, and one rank 3 bug being selected by the players to be fixed the following month.  Rank 4 bugs would be ineligible to vote for given the effort required (something I imagine queuing Battlegrounds as a party as an example would fall under).

From a staff perspective: they are volunteers, they get blamed for everything, and they want to prioritize advancing progression content, so they don't get paid to work, they receive a ton of toxicity, and other areas of the game can go untouched for long periods of time.  Most days, it can truly suck to be a staff member.

From a player perspective, we want responsiveness, we want improvement, and while we want the server to support itself financially we don't want anything that could be construed as pay-to-win.

Currently, I believe the only way a player can influence a bug being fixed falls into 3 categories:
1. raise a shitstorm in public channels (adds to toxicity)
2. fix it yourself (requires specialized knowledge)
3. personally persuade a contributor (non-scalable with a small ratio of contributors-to-players)

With BountySource no longer a viable option, I believe it would be beneficial to take another approach that both encourages engagement with the project (financially) and results in a direct feedback loop to the community.  Someone like Rak who is contributing $30 each month clearly wants this project to succeed and at the same time, CC wants to offer Rak value on his subscription.  I believe by tying subscriptions back to the goal of the project in a way that gives players a voice in partnership is worth considering.  I don't think this will be a single silver bullet that dramatically boosts sub numbers or eliminates all bug frustration, but I do want to see the project keep moving in the right direction.

I'm truly hoping for a discussion that involves both staff and players, with questions and rationales, because Construct has specifically asked for ideas on how to enhance the subscription tiers.

https://www.chromiecraft.com/en/subscriptions-on-chromiecraft/

**Comment by BeefSteaker** [comment]
There used to be a severity/priority system, but it got removed as it didn't really do anything. 

The issue is that contributors are free to work on whatever they want. We don't direct them, and anyone likely to listen to a community priority list likely already knows what the community would want.

**Reply by andovcr** [reply]
Understood Beef.  Effort would be the other axis on this matrix though, would it not?  Effort vs. priority.  I guess my thought is that labeling effort increases transparency to everyone on the complexity of the issue.

And yep, the contributors are volunteers, but it's hard to argue there is not some agreed upon direction already, given that we continue to advance through the progression roadmap, right?  Is the concern that staff members would just ignore whatever the subscribers voted on, in lieu of their own preferences?

**Reply by Nyeriah** [reply]
To evaluate the effort required to solve an issue you would in the first place be required to have the technical knowledge to solve it yourself, which unfortunately is not something we can ask from the people doing the paperwork and helping us maintain the issues triaged. An alternative is to have developers do that but I fail to see how would that contribute to issues getting addressed faster if they are spending their free time doing paperwork instead of addressing the issues themselves.

Furthermore, AzerothCore already uses labels for first time issues/low complexity issues, just not Chromiecraft, for the aforementioned reason. It is not very resource efficient at this stage of the triaging proccess.

It's not particularly helpful to have a list of priority issues set by the public if the list is going to be composed of issues the resident developers were not able to fix. It is well likely that they were not fixed due to reasons such as high complexity/lack of resources rather than lack of interest.

For the same reason I also don't see it as an alternative to Bountysource. Azerothcore is already looking for an alternative to Bountysource, but the purpose of Bountysource was never to fetch the attention of the resident developers but to attract external help to address issues we were not able to address ourselves for one reason or another.
